_posts/2022-10-28-multiple-data-source.md

```diff
@@ -0,0 +1,97 @@
+
+When using this feature, index pattern which are associated with data source will have reference field filled with:
+
+![IndexPattern with dataSource](/assets/media/blog-images/2022-10-28-multiple-data-source/indexpattern-with-ds.png){: .img-fluid}
+
+Current supported data source saved object attributes (pseudo code):
+
+```
+DataSourceAttributes extends SavedObjectAttributes {
+    title,
+    description,
+    endpoint,
+    auth: {
+      type,
+      credentials,
+    };
+}
+```
+
+In supporting other OpenSearch Dashboards’ components to interact with user data source cluster, this feature forked and repurposed existing `opensearch_service` to allow making a data source connection. When OpenSearch Dashboards needs to access data sources on behalf of the user, it will fetch the data source info from its metadata storage by data source id, then send the request to the corresponding data source endpoint.
+
+How to use the new data source client:
+```ts
+client: OpenSearchClient = await context.dataSource.opensearch.getClient(dataSourceId);
+//Support for legacy client
+apiCaller: LegacyAPICaller = context.dataSource.opensearch.legacy.getClient(dataSourceId).callAPI;
+```
+In this phase of development, several OpenSearch Dashboards’ plugins are integrated with the data source concept,  including data, index pattern management, discover and visualization, to allow users to specify a data source when using their functions.
+
+
+**High-level sequence diagram**
+
+![High level sequence diagram](/assets/media/blog-images/2022-10-28-multiple-data-source/highlevel-sequence.png){: .img-fluid}
+
+We plan to provide a more specific design breakdown commit to the OpenSearch Dashboards’ repo. 
+
+
+###What’s next?
+
+With this experimental launch, we are going to continue working on the enhancement including but not limit to:
+
+
+* Other OpenSearch Dashboards plugin support: The multiple data sources currently only works with index pattern based visualizations, we plan to extend to further support with other plugins, and launch data source capability related experience within other plugins.
+* OpenSearch clients will support more authN approaches, such as AWS Sigv4 and JWT.
+* Improve version compatibility: We want to improve the compatibility between the OpenSearch Dashboards and its compatible data sources.
+
+
+
+###Get started with Multi Data Source
+
+For more information on setting up and explore this feature see Data Source Documentation (https://opensearch.org/docs/latest/dashboards/discover/multi-data-sources/). You can try out this feature in OpenSearch playground. To leave feedback, visit OpenSearch Forum (https://forum.opensearch.org/t/feedback-experimental-feature-connect-to-external-data-sources/11144). If you are interested in this topic and want to put your knowledge to further practice, consider contributing to the *OpenSearch-Dashboards* repo!
```

Comment:
1. Suggested revision: 

For more information on setting up and exploring this feature, see the OpenSearch [documentation](https://opensearch.org/docs/latest/dashboards/discover/multi-data-sources/). You can try out this feature in your local environment or the [OpenSearch playground](https://playground.opensearch.org/app/home#/). To leave feedback, visit the OpenSearch Forum. If you're interested in contributing to this experimental feature, consider contributing to the OpenSearch Dashboards repository.